### PR TITLE
fix(web): Sub Article language toggle

### PR DESCRIPTION
**Pull request by RunarVestmann**
# Sub Article language toggle

## What

* Previously clicking on the language toggle button when viewing a sub article would load up the english slug (which is a deprecated field) instead of the english url
* I made a special case for subArticles when the language gets toggled so that you actually visit the correct page now by also fetching the url of an entry when getting content slugs from the backend

### Before

https://user-images.githubusercontent.com/43557895/171065824-dc2770f5-81a7-4c65-91de-572342ec2f49.mp4

### After

https://user-images.githubusercontent.com/43557895/171065926-b8e67ad1-b964-40bc-b0a5-622b7ac4ddc1.mp4

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/web/components/LanguageToggler/LanguageToggler.tsx [hunk 1]
```diff
@@ -71,44 +73,58 @@ export const LanguageToggler = ({
...
+
+    // A subArticle has a url field instead of a slug field and the url field contains slashes
+    if (responses?.[1]?.data?.getContentSlug?.type === 'subArticle') {
+      const slug = responses[1].data.getContentSlug?.url
+      if (slug) {
+        title = responses[1].data.getContentSlug.title
+        type = responses[1].data.getContentSlug.type
+        const slugsIS = slug.is.split('/')
+        const slugsEN = slug.en.split('/')
+        for (let i = 0; i < slugsIS.length && i < slugsEN.length; i += 1) {
```
Comment (kristofer-lp8f) [inline apps/web/components/LanguageToggler/LanguageToggler.tsx:87]: Possible bug if `slugIS.length != slugEn.length`

**Review by kristofer-lp8f — COMMENTED** [review]
Something here rubs me the wrong way. The router should be able to handle slashes by itself. I _really_ don't like indexed for loops.

Do you mind waiting waiting with this until after the weekend?

File: apps/web/components/LanguageToggler/LanguageToggler.tsx [hunk 2]
```diff
@@ -13,8 +13,10 @@ import { Locale } from '@island.is/shared/types'
 import { GET_CONTENT_SLUG } from '@island.is/web/screens/queries/Article'
 import { GlobalContext } from '@island.is/web/context'
 import {
+  ContentSlug,
```
[the remark this answers is not in the document]
Comment (RunarVestmann) [reply apps/web/components/LanguageToggler/LanguageToggler.tsx]: I removed it

File: apps/web/components/LanguageToggler/LanguageToggler.tsx [hunk 3]
```diff
@@ -71,44 +73,58 @@ export const LanguageToggler = ({
...
-        break
+    let title: TextFieldLocales = { is: '', en: '' }
+    let type = ''
+
+    // A subArticle has a url field instead of a slug field and the url field contains slashes
+    if (responses?.[1]?.data?.getContentSlug?.type === 'subArticle') {
+      const slug = responses[1].data.getContentSlug?.url
+      if (slug) {
+        title = responses[1].data.getContentSlug.title
+        type = responses[1].data.getContentSlug.type
```
[the remark this answers is not in the document]
Comment (RunarVestmann) [reply apps/web/components/LanguageToggler/LanguageToggler.tsx:83]: What do you mean exactly, is it the fact that I've got question marks for every field? Would you rather like it to only include question marks before the indexing or just none at all and have a length check before?

**Comment by RunarVestmann** [comment]
This PR can wait, the issue regarding the subarticle language toggle has been around for quite some time now (ever since the PR that deprecated the slug field on subarticles). Feel free to comment on better approaches, I basically just also returned the url also and parsed it via an indexed for loop to split on the slashes (made that specific for the subarticle type).

File: apps/web/components/LanguageToggler/LanguageToggler.tsx [hunk 4]
```diff
@@ -93,7 +93,6 @@ export const LanguageToggler = ({
       }
     }
 
-    // If the subArticle either wasn't there
```
[the remark this answers is not in the document]
Comment (RunarVestmann) [reply apps/web/components/LanguageToggler/LanguageToggler.tsx]: The comment wasn't complete so I removed it

File: apps/web/components/LanguageToggler/LanguageToggler.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (RunarVestmann) [reply apps/web/components/LanguageToggler/LanguageToggler.tsx]: I've added a new commit

[the remark this answers is not in the document]
Comment (RunarVestmann) [reply apps/web/components/LanguageToggler/LanguageToggler.tsx]: Added another commit to address this even better

File: apps/web/components/LanguageToggler/LanguageToggler.tsx [hunk 5]
```diff
@@ -70,45 +71,53 @@ export const LanguageToggler = ({
 
     const responses = await Promise.all(queries)
 
+    const secondContentSlug = responses[1]?.data?.getContentSlug
+
+    // We need to have a special case for subArticles since they've got a url field instead of a slug field
+    if (secondContentSlug?.type === 'subArticle') {
+      const urls = secondContentSlug.url[otherLanguage].split('/')
+
+      if (!secondContentSlug?.title?.[otherLanguage] || urls.length < 2) {
```
Comment (kristofer-lp8f) [inline apps/web/components/LanguageToggler/LanguageToggler.tsx:80]: Can you add a comment on why you have the `.length < 2` check?

**Review by kristofer-lp8f — CHANGES_REQUESTED** [review]
Looks good.

Minor comment, then approval

File: apps/web/components/LanguageToggler/LanguageToggler.tsx (hunk 5 above)
Comment (RunarVestmann) [reply apps/web/components/LanguageToggler/LanguageToggler.tsx:80]: Ok, will do

**Review by kristofer-lp8f — APPROVED** [review]

**Pull request merged**